lib/Service/SnoozeService.php

```diff
@@ -209,41 +187,34 @@ public function unSnoozeMessageDB(Message $message): void {
 	 *
 	 * @throws Exception
 	 */
-	public function snoozeThreadDB(Message $selectedMessage, int $unixTimestamp, Mailbox $srcMailbox): void {
-		$messages = $this->threadMapper->findMessageIdsByThreadRoot(
-			$selectedMessage->getThreadRootId()
-		);
-
-		foreach ($messages as $message) {
+	public function snoozeThreadDB(array $uids, Mailbox $dstMailbox, int $unixTimestamp, Mailbox $srcMailbox): void {
+		foreach ($uids as $uid) {
 			$snooze = new MessageSnooze();
-			$snooze->setMessageId($message['messageId']);
+			$snooze->setMailboxId($dstMailbox->getId());
+			$snooze->setUid($uid);
 			$snooze->setSnoozedUntil($unixTimestamp);
 			$snooze->setSrcMailboxId($srcMailbox->getId());
 
-			$messageId = $message['messageId'];
-
-			$this->messageSnoozeMapper->deleteByMessageIds([$messageId]);
+			// TODO: why delete, then insert?!
```

Comment:
Because we do not have the unique index anymore. And that was the quickest way to make sure that there was just one entry for a message :sweat_smile: .
But if we put an index again on the combi of uid and mailboxId we can catch the unique exception again, I think.
What do you think?